src/main/jbake/content/blog/2017/07/opentracing-ejb.adoc

```diff
@@ -0,0 +1,393 @@
+= OpenTracing EJB instrumentation
+Juraci Paixão Kröhling
+2017-07-28
+:icons: font
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, apm, tracing, jaeger, opentracing, ejb
+
+OpenTracing features more and more framework integrations, allowing for transparent instrumentation
+of applications with minimal effort. This blog post will show how to use the EJB instrumentation
+to automatically trace EJB invocations.
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-28-teaser.png[Jaeger with EJB traces]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-07-28-teaser.png[Jaeger with EJB traces]
+endif::[]
+
+For this demo, we'll generate a project using the Wildfly Swarm project generator, which allows us to
+have a seed project with the appropriate OpenTracing in place. The concrete implementation will be
+Jaeger, which is also provided as a Wildfly Swarm Fraction.
+
+With that, we'll create a simple JAX-RS endpoint with an EJB facet, invoking a set of EJB services in
+different ways to demonstrate all the features of this integration.
+
+Our application has one endpoint called `/order`, responsible for receiving requests to place orders
+in our system. When this endpoint is called, we call a few EJB services, like:
+
+* `AccountService` to send a notification that an order has been placed
```

Comment:
Yeah, I just didn't want to have a method that would return something, so, I thought about notifying that a new order has been placed using that account. If that's bad, I'll change it to `update account's "last activity date" field`